### ISSUE TITLE
Unable to train RNN models with BucketIterator using TPUs

### ISSUE THREAD

**Issue by StephennFernandes** [pytorch/xla#2795]
the Pytorch BucketIterator is very effective to batch samples of similar length, thus minimizing padding and optimizing RNN training. 

But unfortunately `BucketIterator` class doesn't have a `sampler` args in them, sampler would help set `DistributedDataSampler` for TPU devices to do distributed training. 

I tried to emulate  Bucketing similar length functionality by using the bucket_sampler args and collate_fn args in DataLoader, 

but unfortunately the [sampler, bucket_sampler, batch_size] are mutually exclusive, so i use bucket_sampler args i cannot use sampler args , thus restricting me from using TPU distributed training. because i cannot set the `DistributedDataSampler` object to sampler.

**Comment by taylanbil** [comment]
If I understand correctly, this seems like a `BucketIterator` related issue, not torch_xla specific. Sounds like you would have the same issue for distributed GPU/CPU training as well.

I can't follow the part w/ `DataLoader` + `collate_fn` + `DistributedDataSampler`. That combination should work for the bucketing problem in hand. Could you explain the problem you're having in that part?

**Comment by StephennFernandes** [comment]
@taylanbil , currently BucketIterator class doesnt support sampler arguments so i am unable to set a DistributedDataSampler object to  BucketIterator instance , thus it doesn't allow me to train my model on TPUs or any multiple GPUs . 

Please could show me an example where i can train a simple seq2seq or text classification RNN model with TPUs using pytorch and torchtext DataLoader classes.

**Comment by StephennFernandes** [comment]
Hey this issue is solve for me by the torchtext team. thanks for your concern btw.
